connectors/runner.py

```diff
@@ -180,16 +180,17 @@ async def poll(self, one_sync=False, sync_now=True):
             while self.running:
                 try:
                     logger.debug(f"Polling every {self.idling} seconds")
+
                     async for connector in self.connectors.get_connectors(
                         native_service_types, connectors_ids
                     ):
                         await self._one_sync(connector, es, sync_now)
-                    if one_sync:
-                        break
                 except Exception as e:
                     logger.critical(e, exc_info=True)
                     self.raise_if_spurious(e)
-
+                finally:
+                    if one_sync:
```

Comment:
Curious why it fails. Is it because some exceptions are raised and it never reach `if one_syc`?

Moving it to `finally` will ensure the check is done in each iteration of while looo?

Reply:
yes exactly, I had an error in my MySQL connectors, and the raise bypassed the call